src/vs/workbench/contrib/workspace/browser/workspace.contribution.ts

```diff
@@ -77,32 +79,44 @@ export class WorkspaceTrustRequestHandler extends Disposable implements IWorkben
 		@IStatusbarService private readonly statusbarService: IStatusbarService,
 		@IStorageService private readonly storageService: IStorageService,
 		@IWorkspaceTrustRequestService private readonly workspaceTrustRequestService: IWorkspaceTrustRequestService,
+		@IWorkbenchEnvironmentService private readonly workbenchEnvironmentService: IWorkbenchEnvironmentService,
+		@IRemoteAuthorityResolverService private readonly remoteAuthorityResolverService: IRemoteAuthorityResolverService,
 		@IBannerService private readonly bannerService: IBannerService,
 		@IHostService private readonly hostService: IHostService,
 	) {
 		super();
 
 		this.statusbarEntryAccessor = this._register(new MutableDisposable<IStatusbarEntryAccessor>());
 
-		if (isWorkspaceTrustEnabled(configurationService)) {
-			this.registerListeners();
-			this.createStatusbarEntry();
+		(async () => {
 
-			// Set empty workspace trust state
-			this.setEmptyWorkspaceTrustState();
+			await this.workspaceTrustManagementService.workspaceTrustInitialized;
 
-			// Show modal dialog
-			if (this.hostService.hasFocus) {
-				this.showModalOnStart();
-			} else {
-				const focusDisposable = this.hostService.onDidChangeFocus(focused => {
-					if (focused) {
-						focusDisposable.dispose();
-						this.showModalOnStart();
-					}
-				});
+			// Workaround until isTrusted from resolver is available pre-resolution
+			if (this.workbenchEnvironmentService.remoteAuthority) {
```

Comment:
Not sure I understand the value of calling `resolveAuthority()` here.

Reply:
this is to avoid attempting to display any workbench indicators or prompts before we know if the resolver will "lock-in" a trust state. as we discussed, if we can, we should change isTrusted from the resolver to be accessible before resolving the authority. in that case, we will know that workspace trust can[not] be set and we should [not] display prompts or workbench indicators